### PR TITLE
Python 3 support

### PR DESCRIPTION
**Pull request by samuelpowell**
- Enable module build under Python 3
 - Update example script

### CODE REVIEW

**Comment by samuelpowell** [comment]
`make python` now works on Linux under 3.7.3 and 2.7 and `examples/python/recon1.py` (or `recon1_py2.py` resp.) is tested and working. One exception, which I fixed manually, is that the 
 Makefile include has some hard-coded paths for python 2.6. 

@mschweiger my make-foo is poor, is this something you could fix? Do we also need to fix anything for distribution packages/installation?

**Comment by samuelpowell** [comment]
NB this PR is on top of #66 as I needed that to build

**Comment by cboulay** [comment]
Thanks for this. I made a wheel for CPython 3.8 Windows amd64 (attached). I haven't tested all of the functionality yet, but the module imports and I can create a `toast.Mesh` object.

[PyToast-120529-cp38-cp38-win_amd64.zip](https://github.com/toastpp/toastpp/files/6809827/PyToast-120529-cp38-cp38-win_amd64.zip)

**Comment by cboulay** [comment]
To build in Windows:

1. Open the win\vs2015\toast.sln (I use VS2017 but chose No Update)
2. Change the Config to Release and Platform to x64
3. Rt-click and build each of the Mesa\ targets. These need to be built before the top-level targets.
4. Use the Build menu to Build Solution, or build only the targets you want.

Then to generate the Python wheel, from a VS2015 or VS2017 native tools command prompt:

```
conda activate <my_env_with_numpy>
cd src\python
python setup.py build --build-base=../../win/x64/Release/python
python setup.py bdist_wheel
```

The wheel ends up in src\python\dist. It can be installed with `pip install <path\to\PyToast....whl>`. It's portable and can be installed to any other python environment with the same python version (e.g. 3.8) and platform (Win amd64).

Installing the wheel puts the requisite dll files into the conda env's top-leve `site-packages` directory, which is not ideal, but I couldn't figure out how to get the dlls into the wheel with `package_data`, so I used `data_files`.

**Comment by cboulay** [comment]
I'm posting my story in case it helps anyone else out.
tl;dr: When creating a mesh from data, the elements must be 0-based, and the arrays must be C_CONTIGUOUS.

The Python wheel runs the `recon1.py` script well enough. I was also able to run the tutorial scripts after some minimal edits. By the way, now that the wheel and egg includes the binaries, I'm finding that there's no need to do the manual ptoast_install.py step.

I noticed none of the examples create a mesh from data; they all read the mesh from a file. When I attempted to create a Mesh from data using identical matrices (nodes, elems, elem_types) to those that worked in Matlab, it didn't work in Python. I looked at the code and realized that Matlab was assuming 1-based indices in `elems` and Python was assuming 0-based. OK easy enough to decrement the elems array.

This prevented the access violation crash, but it still wasn't working. I was getting negative ElementSize() and the order of magnitude was way off. The nested loops for constructing the node list and constructing the mesh were inverted in the Matlab mex code (mtMesh.cc) compared to the Python extension code (toastmodule.cc `toast_make_mesh`). I modified the Python code to resemble the Matlab code (except the Python code still assumes elems is 0-based) and it worked, yielding identical element sizes. However, this isn't expected because numpy arrays are row-major and matlab is column-major, so the loop nesting should be inverted... **by default**. 

It turns out that the matrices that I saved out from Matlab were imported in Python as F_CONTIGUOUS. Of course. Sigh. With a relayout of the nodes and the elements to order="C", we're back on track.

I suppose either C_CONTIGUOUS or F_CONTIGUOUS case could be handled in the extension code but I'm not good enough with Python.h and numpy/arrayobject.h to handle that.

**Comment by samuelpowell** [comment]
Included in #72

**Comment by rob-luke** [comment]
🎉 🎉 🎉 🎉 🎉 🎉

**Comment by samuelpowell** [comment]
@rob-luke don’t get too excited. A lot more required to make this viable long term. 

I do have a plan. But it would be useful to know what license would be required for mne-nirs / mne before investing much time. 

Would LGPL work for example?
